### PR TITLE
Ft add user rest api 149466927

### PR DESCRIPTION
**Pull request by dmwiine**
## PR Status
**READY**

## Story ID
149466927

## Added Migrations
YES 

## Description
Allow creation of users via REST API. External apps might need to register accounts on behalf of their users without having to send them over to the wger instance.

## New Packages Added
None

## Todos
- [ ] Tested and working locally
- [ ] Tested and working on development environment
- [ ] Added more Unit tests (if appropriate)
- [ ] Reviewed and approved by a senior dev
- [ ] Reviewed and approved by the scrum master

## Impacted Areas in Application
List general components of the application that this PR will affect:
* core

### CODE REVIEW

**Comment by johnseremba** [comment]
@dmwiine @charlieoduk The pull request entails too many commits and very many changes that were not possibly done by you. 
It's hard to know what exactly you changed.

File: wger/core/tests/test_create_user_via_api.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            username='FitnessFirst',
+            email='email@oduk.com',
+            password='password')
+        user.save()
+        profile = UserProfile.objects.get(user=user)
+        profile.can_create_via_api = True
+        profile.save()
+
+        self.payload = {
+            'user_id': user.id,
```
Comment (johnseremba) [inline wger/core/tests/test_create_user_via_api.py]: This part doesn't seem right.
`user_id: user.id` is using the user `FitnessFirst`'s ID to create a new user `DonnaMwiine`.
I don't think it is necessary to pass the `user.id` in the api payload.

File: wger/core/tests/test_create_user_via_api.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        '''
+        Tests the creation of a user via the API
+        '''
+        user = User.objects.create_user(
+            username='FitnessFirst',
+            email='email@oduk.com',
+            password='password')
+        user.save()
+        profile = UserProfile.objects.get(user=user)
+        profile.can_create_via_api = True
```
Comment (johnseremba) [inline wger/core/tests/test_create_user_via_api.py]: I think there is a way to generate an API Key for the user which you can pass in the Headers. I guess that would be a better implementation.

File: wger/core/tests/test_add_user_commands.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            'list-user-rest-api', 'Williams')
+
+        with open('wger/core/tests/test.txt') as f:
+            data = f.readlines()
+        reader = csv.reader(data)
+        william_users = []
+        for row in reader:
+            william_users.append(row)
+        self.assertIn('Another Api Man', william_users[1])
+        self.assertIn('Api Man', william_users[0])
```
Comment (johnseremba) [inline wger/core/tests/test_add_user_commands.py:70]: @dmwiine @charlieoduk Did you guys create this file??

File: wger/core/api/views.py [hunk 4]
```diff
@@ -33,12 +35,37 @@
...
 from wger.core.api.serializers import UserprofileSerializer
 from wger.utils.permissions import UpdateOnlyPermission, WgerPermission
 
 
+class UserViewSet(viewsets.ModelViewSet):
+    serializer_class = UserCreationSerializer
+    queryset = User.objects.all()
+
+    def create(self, request):
+        user = User.objects.get(id=str(loads(request.body.decode())['user_id']))
```
Comment (johnseremba) [inline wger/core/api/views.py]: I feel this `user_id` can be decoded from the API Key if sent as a header.

**Review by johnseremba — COMMENTED** [review]
Good work guys. I have added some comments.
When you rebased, you added so many files which you didn't change, including migrations.

File: wger/core/tests/test_add_user_commands.py (hunk 3 above)
Comment (charlieoduk) [reply wger/core/tests/test_add_user_commands.py:70]: @SerryJohns Yes we did

File: wger/core/api/views.py (hunk 4 above)
Comment (charlieoduk) [reply wger/core/api/views.py]: Ok, we'll look into that implementation

**Review by johnseremba — APPROVED** [review]
Looks good to me.

**Review by collin5 — APPROVED** [review]
Look ok to me, good work

**Review by amwaleh — APPROVED** [review]

**Pull request merged**